src/geanyobject.c

```diff
@@ -156,6 +156,12 @@ static void create_signals(GObjectClass *g_object_class)
 		G_SIGNAL_RUN_FIRST,
 		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
 		G_TYPE_NONE, 0);
+	geany_object_signals[GDB_PROJECT_BEFORE_CLOSE] = g_signal_new (
+		"project-before-close",
+		G_OBJECT_CLASS_TYPE (g_object_class),
+		G_SIGNAL_RUN_FIRST,
+		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
+		G_TYPE_NONE, 0);
```

Comment:
Wouldn't it be better if the signal received the `GeanyProject` as argument?
OK, it'd be inconsistent with the other project signals, so maybe it's a bad idea. Or maybe it's a good idea to break one less thing in the future if we want to allow more than one project to be open?  That might not make much sense because the other signals would have to be changed anyway, so meh.